### DISCUSSION TITLE
Configuring Web Proxy for OpenAiProvider

### DISCUSSION THREAD

**Discussion by code-monkey88** [tryAGI/LangChain#430] [Q&A] [answered]
Is there a way to set up a web proxy by specifying an HttpClient when generating an OpenAiProvider?

**Comment by HavenDV** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/dc968c99-eefb-46ff-bd0d-ab4b99fd9607)
You can use this constructor and pass HttpClient inside OpenAiApi

![image](https://github.com/user-attachments/assets/458a7881-f601-4c6b-b16d-14b61f90eec4)

**Reply by code-monkey88** [reply]
I was able to achieve this using the following constructor, based on the answers you provided.
Thank you very much.

![image](https://github.com/user-attachments/assets/6855340f-8739-4281-bd60-e50d680e5fe9)

![image](https://github.com/user-attachments/assets/95a31431-a1e6-43b4-896a-cbdd9303f5e6)
